spyder_terminal/widgets/terminalgui.py

```diff
@@ -179,11 +186,18 @@ def contextMenuEvent(self, event):
         event.accept()
 
     def eval_javascript(self, script):
-        """Evaluate Javascript instructions inside DOM."""
+        """
+        Evaluate Javascript instructions inside DOM with the expected prefix.
+        """
+        script = PREFIX + script
         if WEBENGINE:
-            return self.document.runJavaScript("{}".format(script))
+            self.document.runJavaScript("{}".format(script), self.return_value)
         else:
-            return self.document.evaluateJavaScript("{}".format(script))
+            self.document.evaluateJavaScript("{}".format(script))
+
+    def return_value(self, value):
+        """Returns the value of the function evaluated in javascript."""
```

Comment:
Let's change this docstring to

    Return the value of a function evaluated in Javascript.